sys/include/storage/flash_sim.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2015 Lucas Jenß
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_storage
+ * @brief
+ * @{
+ *
+ * @brief       Simulates flash memory for easier testing of storage subsystems
+ * @author      Lucas Jenß <lucas@x3ro.de>
+ */
+
+#ifndef _FS_FLASH_SIM_H
+#define _FS_FLASH_SIM_H
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Enumerates all error codes returned by the flash simulation layer
+ */
+typedef enum flash_sim_error {
+    E_FS_SUCCESS = 0,            //!< Operation was successful
+    E_FS_INVALID_WRITE = 1,      //!< Trying to set a bit from 0 -> 1 w/ a program operation
+    E_FS_INVALID_PARAMS = 2,     //!< Invalid parameters were passed to the function
+    E_FS_OUT_OF_RANGE = 3,       //!< The operation was out of range
+    E_FS_FILE_ERROR = 4,         //!< Error operating on the file simulating the flash device
+    E_FS_NOT_INITIALIZED = 5     //!< Flash simulation layer was not initialized prior to operation
+} flash_sim_error_t;
+
+/**
+ * @brief Configuration for the flash simulation layer
+ */
+typedef struct flash_sim {
+    uint32_t page_size;         //!< Page size to be emulated
+    uint32_t block_size;        //!< Block size (must be a multiple of the page size)
+    uint64_t storage_size;      //!< Total amount of storage to be made available
+
+    uint32_t pages;             //!< Number of pages (computed from the above)
+    FILE *_fp;                  //!< File descriptor for the file simulating the flash device
```

Comment:
By putting this file into `sys/` this will lead to problems as soon as we support `FILE` pointers on our own. Since this is an emulation layer for native I would propose to move the whole file into `cpu/native` (similar to `netdev_tap`).